src/ui/public/kfetch/kfetch_abortable.ts

```diff
@@ -29,7 +31,10 @@ function createAbortable() {
   };
 }
 
-export function kfetchAbortable(fetchOptions, kibanaOptions) {
+export function kfetchAbortable(
+  fetchOptions?: Omit<KFetchOptions, 'signal'>,
```

Comment:
I'm probably reading this wrong but I don't understand the need for excluding `signal`. It's not part of `KFetchOptions`.

I'd expect `fetchOptions: KFetchOptions` to be fine - but maybe not?

Reply:
`KFetchOptions` extends the default options for the global fetch API, which includes `signal`, which is why I've explicitly removed it from the type here (since we overwrite it)

Reply:
I see that the `RequestInit` interface has `signal` of type `AbortSignal`. The `signal` we are overwriting `kibanaOptions` with is also of type `AbortSignal`

I pulled your branch and removed the `Omit` part, and it still compiles. So sounds ok to me - but haven't worked with TS for a while so I might be missing something.

Do you get a compile error if you remove the Omit part?

Reply:
Nope, the type described in the arguments should be the type that callers will get. I'm just trying to remove the `signal` option so when a caller is looking at the options through intelisense or something they will see `signal` as an option for `kfetch()`:

![image](https://user-images.githubusercontent.com/1329312/42920973-52176656-8ace-11e8-9040-67b1dc87aeb3.png)

but not for `kfetchAbortable()`:

![image](https://user-images.githubusercontent.com/1329312/42921011-7cc4a5b2-8ace-11e8-8ef8-76b21cd1edef.png)

Reply:
Got it :) Thanks for the explanation.